sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -175,6 +197,28 @@ object JDBCRDD extends Logging {
       requiredColumns: Array[String],
       filters: Array[Filter],
       parts: Array[Partition],
+      options: JDBCOptions): RDD[InternalRow] = {
+    val url = options.url
+    val dialect = JdbcDialects.get(url)
+    val quotedColumns = requiredColumns.map(colName => dialect.quoteIdentifier(colName))
+    val predicates = filters.flatMap(toV2(_))
+    new JDBCRDD(
+      sc,
+      dialect.createConnectionFactory(options),
+      pruneSchema(schema, requiredColumns),
+      quotedColumns,
+      predicates,
+      parts,
+      url,
+      options)
+  }
+
+  def scanTable(
```

Comment:
Can we remove the old `scanTable` method now? The caller side can call the new one with a simple change: `filters.map(_.toV2)`